packages/twilio-run/src/runtime/server.ts

```diff
@@ -57,6 +60,99 @@ function requireCacheCleaner(
   next();
 }
 
+async function findRoutes(config: StartCliConfig): Promise<RouteInfo> {
+  const searchConfig: SearchConfig = {};
+
+  if (config.functionsFolderName) {
+    searchConfig.functionsFolderNames = [config.functionsFolderName];
+    console.log(searchConfig);
+  }
+
+  if (config.assetsFolderName) {
+    searchConfig.assetsFolderNames = [config.assetsFolderName];
+  }
+
+  return getFunctionsAndAssets(config.baseDir, searchConfig);
+}
+
+function configureWatcher(config: StartCliConfig, server: LDS) {
+  const watcher = chokidar.watch(
+      [
+        path.join(config.baseDir, config.functionsFolderName ? `/(${config.functionsFolderName})/**/*)` : '/(functions|src)/**/*.js'),
+        path.join(config.baseDir, config.assetsFolderName ? `/(${config.assetsFolderName})/**/*)` : '/(assets|static)/**/*'),
+      ],
+      {
+        ignoreInitial: true,
+      }
+    );
+
+    const reloadRoutes = async () => {
+      const routes = await findRoutes(config);
+      server.update(routes);
+    };
+
+    // Debounce so we don't needlessly reload when multiple files are changed
+    const debouncedReloadRoutes = debounce(reloadRoutes, RELOAD_DEBOUNCE_MS);
+
+    watcher
+      .on('add', path => {
+        debug(`Reloading Routes: add @ ${path}`);
+        debouncedReloadRoutes();
+      })
+      .on('unlink', path => {
+        debug(`Reloading Routes: unlink @ ${path}`);
+        debouncedReloadRoutes();
+      });
+
+    // Clean the watcher up when exiting.
+    process.on('exit', () => watcher.close());
+}
+
+export async function createLocalDevelopmentServer(
+  port: string | number = DEFAULT_PORT,
+  config: StartCliConfig
+): Promise<Express> {
+  try {
+    const { LocalDevelopmentServer } = requireFromProject(
+      config.baseDir,
+      '@twilio/runtime-handler/dev'
+    ) as { LocalDevelopmentServer: LDS };
+
+    const routes = await findRoutes(config);
+
+    const server = new LocalDevelopmentServer(port, {
+      inspect: config.inspect,
+      baseDir: config.baseDir,
+      env: config.env,
+      port: config.port,
+      url: config.url,
+      detailedLogs: config.detailedLogs,
+      live: config.live,
+      logs: config.logs,
+      legacyMode: config.legacyMode,
+      appName: config.appName,
+      forkProcess: config.forkProcess,
+      logger: logger,
+      routes: routes,
+      enableDebugLogs: true
+    });
+    server.on('request-log', (logMessage) => {
+      writeOutput(logMessage)
+    });
+    server.on('updated-routes', async (config) => {
+      await printRouteInfo(config);
+    })
+    configureWatcher(config, server);
+    return server.getApp();
+  } catch (err) {
+    debug(
```

Comment:
Do we need to catch this if the version of `twilio-run` depends on `runtime-handler`? What is the situation in which this version of `twilio-run` would not have access to a `LocalDevelopmentServer` from this package, regardless of which version it is?

Reply:
And consequently, do we need to keep all the server code in `twilio-run` as well as in `runtime-handler`?

Reply:
it doesn't actually depend on runtime-handler. It's a `devDependency` for the type but the require comes from the customers project not bundled from `twilio-run`. We could do both but I feel like that might introduce more edge cases than just ripping the bandaid off in the next major release and dropping all the code.